sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.students.controller.getRequestedAssessmentSummary.e2e-spec.ts

```diff
@@ -0,0 +1,146 @@
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should get the student assessment requests summary for an eligible application for a student when they try to access it.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        student,
+      },
+      { applicationStatus: ApplicationStatus.Completed },
+    );
+    // Create a declined student appeal.
+    const declinedAppealRequest = createFakeStudentAppealRequest();
+    declinedAppealRequest.appealStatus = StudentAppealStatus.Declined;
+    const declinedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [declinedAppealRequest],
+    });
+    await db.studentAppeal.save(declinedAppeal);
+    // Create an approved student appeal.
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    const approvedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [approvedAppealRequest],
+    });
+    await db.studentAppeal.save(approvedAppeal);
+    // Create a pending student appeal.
+    const pendingAppealRequest = createFakeStudentAppealRequest();
+    pendingAppealRequest.appealStatus = StudentAppealStatus.Pending;
+    const pendingAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [pendingAppealRequest],
+    });
+    await db.studentAppeal.save(pendingAppeal);
+    // Create approved application offering change request.
+    await saveFakeApplicationOfferingRequestChange(
+      db,
+      {
+        application,
+      },
+      {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.Approved,
+        },
+      },
+    );
+    // Create declined application offering change request.
+    const declinedApplicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(
+        db,
+        {
+          application,
+        },
+        {
+          initialValues: {
+            applicationOfferingChangeRequestStatus:
+              ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+          },
+        },
+      );
+    // Create pending application offering change request.
+    const pendingApplicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(db, { application });
+    const endpoint = `/students/assessment/application/${application.id}/requests`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([
```

Comment:
just wondering, in the request, there is sort right? pending first and declined next? correct me if I am wrong

Reply:
There is sort for the appeals but not for the application offering change. I didn't add the sort because it wasn't needed for the UI but seems like it would be needed for a deterministic output in the case of e2e tests. Please correct me if I am wrong.